aws/table_aws_wellarchitected_lens.go

```diff
@@ -0,0 +1,243 @@
+				Description: "An Amazon Web Services account ID.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "share_invitation_id",
+				Description: "The ID assigned to the shared invitation.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getWellArchitectedLens,
+				Transform:   transform.FromField("ShareInvitationId"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LensName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getWellArchitectedLens,
+				Transform:   transform.FromField("Tags"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("LensArn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listWellArchitectedLenses(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := WellArchitectedClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_lens.listWellArchitectedLenses", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxLimit := int32(50)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			maxLimit = limit
+		}
+	}
+
+	input := &wellarchitected.ListLensesInput{
+		MaxResults: maxLimit,
+	}
+
+	if d.EqualsQualString("lens_status") != "" {
+		input.LensStatus = types.LensStatusType(d.EqualsQualString("lens_status"))
+	}
+	if d.EqualsQualString("lens_name") != "" {
+		input.LensName = aws.String(d.EqualsQualString("lens_name"))
+	}
+	if d.EqualsQualString("lens_type") != "" {
+		input.LensType = types.LensType(d.EqualsQualString("lens_type"))
+	}
+
+	paginator := wellarchitected.NewListLensesPaginator(svc, input, func(o *wellarchitected.ListLensesPaginatorOptions) {
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_wellarchitected_lens.listWellArchitectedLenses", "api_error", err)
+			return nil, err
+		}
+
+		for _, item := range output.LensSummaries {
+			// As per the doc(https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_LensSummary.html)
+			// The lens alias is same as lens arn if it is a custom lens.
+			// The lens alias value coming as nil in the case of custom lenses, so we are replacing the null value with lens arn.
+			if item.LensAlias == nil || *item.LensAlias == "" {
+				item.LensAlias = item.LensArn
+			}
```

Comment:
@ParthaI Is it valid to alter the column value if the API returns nil? Should we not stick to what the API returns?

Reply:
- As per the doc(https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_LensSummary.html)
- The lens alias is same as lens ARN if it is a custom lens.

If we will run the following query ` select lens_name, lens_alias, lens_version, region from aws_wellarchitected_lens where lens_alias = 'arn:aws:wellarchitected:us-east-1:123456789012:lens/bd49240eee87eb7b6022c2cc93c5b110'` for a custom lens in this case we will get nil value in the column `lens_alias`(API doesn't provide) lens alias value for a custom lens. A filtration will be apply in steampipe level(which is a default behaviour of postgreSQL) which results empty row due to mismatch of API response and the quals value provided in where clause.

Not sure if it is the correct way to do or not.